common/src/main/java/io/crate/types/IntervalType.java

```diff
@@ -0,0 +1,110 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.types;
+
+import io.crate.Streamer;
+import io.crate.interval.IntervalParser;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.joda.time.Period;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.Locale;
+
+public class IntervalType extends DataType<Period> implements FixedWidthType, Streamer<Period> {
+
+    public static final int ID = 17;
+    public static final IntervalType INSTANCE = new IntervalType();
+
+    @Override
+    public int id() {
+        return ID;
+    }
+
+    @Override
+    public Precedence precedence() {
+        return Precedence.IntervalType;
+    }
+
+    @Override
+    public String getName() {
+        return "interval";
+    }
+
+    @Override
+    public Streamer<Period> streamer() {
+        return this;
+    }
+
+    @Override
+    public Period value(Object value) throws IllegalArgumentException, ClassCastException {
+        if (value == null) {
+            return null;
+        }
+
+        if (value instanceof Period) {
+            return (Period) value;
+        }
+
+        if (value instanceof String) {
+            return IntervalParser.apply((String) value);
+        }
+
+        throw new IllegalArgumentException(String.format(Locale.ENGLISH, "Cannot convert %s to interval", value));
+    }
+
+    public int compareValueTo(Period p1, Period p2) {
+        return nullSafeCompareValueTo(p1.toStandardDuration(), p2.toStandardDuration(), Comparator.naturalOrder());
+
+    }
+
+    @Override
+    public Period readValueFrom(StreamInput in) throws IOException {
+        if (in.readBoolean()) {
+            int seconds = Math.toIntExact(in.readLong());
+            int days = in.readInt();
+            int months = in.readInt();
+            return new Period().withSeconds(seconds).withDays(days).withMonths(months);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public void writeValueTo(StreamOutput out, Period p) throws IOException {
+        if (p == null) {
+            out.writeBoolean(false);
+        } else {
+            out.writeLong(p.getSeconds());
+            out.writeInt(p.getDays());
+            out.writeInt(p.getMonths());
+        }
+    }
+
+    @Override
+    public int fixedSize() {
+        return 32;
```

Comment:
Is this still accurate?

Reply:
This is estimated with JOL plugin based `joda.Period`